### ISSUE TITLE
Pointless operator split in OH/Cl methane destruction biases diagnostics

### ISSUE THREAD

**Issue by toddmooring** [geoschem/geos-chem#1738]
### Name and Institution (Required)

Name: Todd Mooring
Institution: Harvard ACMG

### Confirm you have reviewed the following documentation

I affirm that I have read the relevant documentation.

### Description of your issue or question

In the most recent released version of global_ch4_mod.F90 for the methane specialty simulation (commit 1ba0f84) there is a formal inconsistency between the equations used to update the methane concentration in response to destruction by OH and Cl and the diagnostic output about the OH- and Cl-driven destruction rates.

The issue occurs in lines 949-961: lines 949-957 contain the loss rate diagnostics, which are both called on the same value of the methane `GCH4`. The actual updating of `GCH4` in response to the OH- and Cl- driven losses is lines 960-961. The OH-driven loss is computed using the same `GCH4` value underlying the diagnostic calculation, but the Cl-driven loss (line 961) is computed using a `GCH4` value that has already experienced the OH-driven loss and is therefore smaller.

The estimated Cl-driven loss sent to the diagnostic output is therefore larger than the actual value of the Cl-driven loss used to update the model's prognostic methane field. Although tests suggest that this error is of negligible significance for the closure of the globally-integrated methane budget, it still seems worth fixing from a consistency perspective.

This could be done by removing the apparently pointless operator split at lines 960-961 and replacing them with a single new line:

`GCH4 = GCH4 * ( 1e+0_fp - KRATE * C_OH * DT  - KRATE_Cl * C_Cl * DT )`

or similar.

**Comment by yantosca** [comment]
Thanks @toddmooring.  Does this also impact the new carbon simulation with KPP?

**Comment by yantosca** [comment]
@toddmooring @msulprizio:  I did a quick test to make sure the equation above is equivalent:

          ! Calculate new CH4 value: [CH4]=[CH4](1-k[OH]*delt)
          GCH4_old = GCH4 *     ( 1e+0_fp - KRATE    * C_OH * DT )
          GCH4_old = GCH4_old * ( 1e+0_fp - KRATE_Cl * C_Cl * DT )

          GCH4_new = GCH4 *                                                  &
             ( 1.0_fp - ( KRATE * C_OH * DT ) - ( KRATE_Cl * C_Cl * DT )  )

          IF ( L==1  ) THEN
             print*, I, J, GCH4_old, GCH4_new, GCH4_new-GCH4_old
          ENDIF

and I get
```console
          I           J    GCH4_old                  GCH4_new                    GCH4_new - GCH4_old
          39          45   57339795252837.922        57339795252837.922        0.0000000000000000     
          40          45   57375176823570.680        57375176823570.680        0.0000000000000000     
          41          45   57460315285407.297        57460315285407.297        0.0000000000000000     
          42          45   57576903962656.328        57576903962656.320       -7.8125000000000000E-003
          43          45   57693924669924.938        57693924669924.930       -7.8125000000000000E-003
          44          45   57803850643109.773        57803850643109.766       -7.8125000000000000E-003
          45          45   57906062107570.664        57906062107570.664        0.0000000000000000     
          46          45   57984493246791.594        57984493246791.594        0.0000000000000000     
          47          45   58033962325845.797        58033962325845.797        0.0000000000000000     
          48          45   58066618646999.984        58066618646999.984        0.0000000000000000     
          49          45   58089612625242.914        58089612625242.914        0.0000000000000000     
          50          45   58141206655500.539        58141206655500.539        0.0000000000000000     
          51          45   58175507286579.430        58175507286579.430        0.0000000000000000     
          52          45   58201104769504.008        58201104769504.008        0.0000000000000000     
          53          45   58213448132551.109        58213448132551.109        0.0000000000000000     
          54          45   58244692095375.133        58244692095375.133        0.0000000000000000     
          55          45   58297315513394.609        58297315513394.609        0.0000000000000000     
          56          45   58265058032669.977        58265058032669.969       -7.8125000000000000E-003
          57          45   58234556044008.008        58234556044008.008        0.0000000000000000     
```
(that is a small subset of the data but you get the idea).

The differences are either 0 exactly or  -7e-3, which I think may be introduced by the compiler optimizations.  It is essentially zero but will result in small differences in the output.

**Comment by yantosca** [comment]
A few boxes have a little higher difference but still it is essentially the same to many decimal places:
```console
          33          41   54547080906322.805        54547080906322.789       -1.5625000000000000E-002
          34          41   54615063793060.992        54615063793060.969       -2.3437500000000000E-002
          35          41   54203106638921.312        54203106638921.297       -1.5625000000000000E-002
          36          41   53386135710750.938        53386135710750.914       -2.3437500000000000E-002
          37          41   52618817976875.961        52618817976875.938       -2.3437500000000000E-002
          38          41   51834032037094.852        51834032037094.828       -2.3437500000000000E-002
          39          41   51217146016849.156        51217146016849.141       -1.5625000000000000E-002
          40          41   50912410407343.391        50912410407343.375       -1.5625000000000000E-002
          41          41   50622088566262.523        50622088566262.516       -7.8125000000000000E-003
```

**Comment by toddmooring** [comment]
Thanks for looking into this. I haven't tried to use the new carbon simulation so I haven't looked at its code at all. By the "this" in your question do you mean the operator split/diagnostics problem or the loss frequencies matter?

-Todd
________________________________

**Comment by msulprizio** [comment]
The pull request (#1740) for this issue has now been merged into the 14.2.0 development branch.
